test/sharness/t0080-repo.sh

```diff
@@ -152,6 +152,31 @@ test_expect_success "'ipfs pin ls --type=all --quiet' is correct" '
 	test_sort_cmp allpins_uniq_hashes actual_allpins
 '
 
+test_expect_success "'ipfs refs --unique' is correct" '
+	mkdir -p uniques &&
+	cd uniques &&
+	echo "content1" > file1 &&
+	echo "content1" > file2 &&
+	ROOT=`ipfs add -r -q . | tail -n 1` &&
+	ipfs refs --unique $ROOT > expected &&
+	ipfs add -q file1 > unique_hash &&
+	test_cmp expected unique_hash
+'
+
+test_expect_success "'ipfs refs --unique --recursive' is correct" '
+	rm * &&
+	mkdir -p sub1/sub2 &&
+	echo "content1" > file1 &&
+	echo "content1" > sub1/file1 &&
+	echo "content1" > sub1/sub2/file1 &&
+	echo "content2" > sub1/sub2/file2 &&
+	ROOT=`ipfs add -r -q . | tail -n 1` &&
+	ipfs refs --unique --recursive $ROOT > refs_output &&
+	wc -l refs_output > line_count &&
+	echo "4 refs_output" > expected &&
+	test_cmp line_count expected
+'
+
```

Comment:
maybe we could have 1 more test with a larger hierarchy? i'm always skeptical of small tests. maybe: 
```sh
mkdir -p a/b/c/d &&
echo "content1" >a/f &&
echo "content1" >a/b/f1 &&
echo "content1" >a/b/c/f2 &&
echo "content2" >a/b/f2 &&
echo "content2" >a/b/c/f1 &&
echo "content2" >a/b/c/d/f &&
cp -r a a2 && mv a2 a/a2 &&
cp -r a a3 && mv a3 a/a3 &&
cp -r a a4 && mv a4 a/a4
```

then can test with things like:
```sh
ipfs refs -r $root >all
ipfs refs -r -u $root >unique
# and
ipfs refs -r -u $root >unique_actual_sorted &&
ipfs refs -r $root | sort | uniq >unique_expected &&
test_cmp unique_expected unique_actual_sorted 
```